js/queries.js

```diff
@@ -51,14 +58,18 @@ function getSourceText(state, actor) {
 }
 
 function getBreakpoint(state, location) {
+  location = Immutable.fromJS(location);
   return getBreakpoints(state).get(makeLocationId(location));
 }
 
 /**
  * @param object - location
  */
 function makeLocationId(location) {
-  return location.actor + ":" + location.line.toString();
+  if(!Immutable.Iterable.isIterable(location)) {
+    location = Immutable.fromJS(location);
+  }
+  return location.get("actor") + ":" + location.get("line").toString();
```

Comment:
I could see queries and actions being immutable as well. 
+ The argument for reducers is that way they receive and set immutable objects. 
+ Queries is less clear, but at this point i believe they mostly receive a string or immutable object

With that said, I'd be fine with actions, queries, and client calls receiving plain JS objects. I haven't thought this through and you have more experience here.